src/main/java/org/apache/maven/plugins/clean/Cleaner.java

```diff
@@ -336,6 +341,15 @@ private int delete(Path file, boolean failOnError, boolean retryOnError) throws
         return 0;
     }
 
+    private static IOException delete(Path file) {
+        try {
+            Files.deleteIfExists(file);
+        } catch (IOException e) {
+            return e;
```

Comment:
This is an unidiomatic pattern for Java. I would expect this to simply throw the exception and have that case handled in a catch block.

Reply:
Maybe can be designed in better way, but please consider that it is moved code from 3.x branch to master.